core/fetcher.go

```diff
@@ -177,3 +177,34 @@ func (f *BlockFetcher) UnsubscribeNewBlockEvent(ctx context.Context) error {
 
 	return f.client.Unsubscribe(ctx, newBlockSubscriber, newBlockEventQuery)
 }
+
+// IsSyncing returns the sync status of the Core connection: true for
+// syncing, and false for already caught up. It can also return an error
+// in the case of a failed status request.
+func (f *BlockFetcher) IsSyncing(ctx context.Context) (bool, error) {
+	resp, err := f.client.Status(ctx)
+	if err != nil {
+		return false, err
+	}
+	return resp.SyncInfo.CatchingUp, nil
+}
+
+// WaitFinishSync will block until the Core connection has finished syncing,
+// returning a nil error.
```

Comment:
Can this document in which cases an error will be returned? Just reading the comment it looks like it will never return an error!

Reply:
resolved?